### ISSUE TITLE
Wrong js error checking for decimal numbers

### ISSUE THREAD

**Issue by gpoehl** [yiisoft/yii2#8407]
I habe an activeForm field using MaskedInput widget with this client options

``` php
'clientOptions' => [
        'alias' =>  'decimal',
       'groupSeparator' => '.',
        'radixPoint' => ',', 
        'autoGroup' => true
    ]
```

and a this rule defined for the field `[['ablesewert',], 'number'],`
After editing the filed I get the error message : **Ablesewert is not a number**.

How can I achive the goal to show and edit a decimal field using a redixPoint = ','?

**Comment by cebe** [comment]
you have to configure the http://www.yiiframework.com/doc-2.0/yii-validators-numbervalidator.html#$numberPattern-detail to match your locale.

**Comment by cebe** [comment]
this could be taken as an enhacement request, the validator could respond to application locale settings itself, using intl.

**Comment by nkovacs** [comment]
Only allowing the locale's decimal separator would break websites that use a locale with `,`, but didn't change the number format from the default - loading existing data with a `.` into a form would not allow you to edit it after upgrading.

You could change the default numberPattern to allow both `.` and `,`. This wouldn't break BC, and it would improve usability for users who use a keyboard with a `,` on the numpad and wish to enter a number on a website that's set to a locale that uses `.`. The only possible issue is someone entering a thousands separator `,` without a decimal separator `.`, which would then be interpreted as a decimal separator `,`.

There's also the problem that php can only convert strings with `.` as a decimal separator to floats, and will actually cut off the decimals (`(float)'8,7' === 8`), and a decimal comma will probably also cause problems with database insertions. So without proper conversion, the value would pass validation, only to fail at a later stage or cause data loss.

The validator also shouldn't change the value, because in case of a validation error in another attribute during submission, the user will get the form back with a decimal point instead of a decimal comma. You could use a separate attribute for the converted value (like `timestampAttribute` in DateValidator), but that would complicate the model. I think a possible solution to this and the `timestampAttribute` would be to allow a validator to modify the data after validation and before database insertion. This would make working with dates and times easier as well. You could also add a function to modify the value before displaying it in a form, so that a unix timestamp or a number with a decimal point could be converted automatically to a formatted datetime or a number with a decimal comma before displaying it to the user. Currently you have to do this manually.

**Comment by thiagotalma** [comment]
Any news here?

This improvement would be very good!
I am suffering with this issue now.

**Comment by samdark** [comment]
Won't be implemented in 2.0.
